rcl_action/include/rcl_action/action_client.h

```diff
@@ -742,6 +744,43 @@ bool
 rcl_action_client_is_valid(
   const rcl_action_client_t * action_client);
 
+/// Configures service introspection features for all internal service clients of the action client.
+/**
+ * For the internal goal, cancel, and result clients of the action client, call
+ * \ref rcl_client_configure_service_introspection separately for each.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | Maybe [1]
+ * Lock-Free          | Maybe [1]
+ * <i>[1] rmw implementation defined</i>
+ *
+ * \param[in] action_client action client on which to configure internal service introspection
+ * \param[in] node valid rcl_node_t to use to create the introspection publisher
+ * \param[in] clock valid rcl_clock_t to use to generate the introspection timestamps
+ * \param[in] type_support type support library associated with this action client
+ * \param[in] publisher_options options to use when creating the introspection publisher
+ * \param[in] introspection_state rcl_service_introspection_state_t describing whether
+ *            introspection should be OFF, METADATA, or CONTENTS
+ * \return #RCL_RET_OK if the call was successful, or
+ * \return #RCL_RET_ERROR if calling rcl_client_configure_service_introspection for each internal
+ *         client doesn't return RCL_RET_OK, or
+ * \return #RCL_RET_INVALID_ARGUMENT if the given node or clock or type_support is invalid
+ */
+RCL_ACTION_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_action_client_configure_internal_service_introspection(
```

Comment:
i would name like this.

```suggestion
rcl_action_client_configure_action_introspection(
```

the reason is, from user's perspective this actually enables action introspection. i understand that it internally enables the service (action hidden services) introspection, but that is not really clear for user's point of view. if we change the name, doc section needs to be updated as well.

Reply:
Agree. I will update.